### PR TITLE
add exercise for running check()

### PR DESCRIPTION
**Pull request by f-hafner**
working on https://github.com/carpentries-incubator/lesson-R-packaging/issues/81

from [week 3](https://github.com/esciencecenter-digital-skills/2022-09-14-ds-rpackaging/blob/main/files/CollaborativeDocument-week3.md), add the exercise on running `check()`. I created a small own exercise so that people can either do this on their own package, or in the example `mysterycoffee` package. I also added some of the discussion about the pipe operator from [here](https://github.com/esciencecenter-digital-skills/2022-09-14-ds-rpackaging/blob/main/files/CollaborativeDocument-week2.md). Perhaps this discussion belongs elsewhere?

### CODE REVIEW

File: _episodes/07-dependencies.md [hunk 1]
```diff
@@ -90,4 +90,64 @@ More specifically, we can see that a version equal or higher than `3.0.0` is sug
...
+>  df <- data.frame(make_groups(names))
+>  names(df) <- c("group1", "group2")
+>  out <- df %>% tidyr::pivot_longer(cols = tidyselect::starts_with("group"))
+>  return(out)
+>}
+>~~~
+> {: .code}
+> 
+> With this in place, or with your own package, do the following.
+> Run `devtools::check()` (or `Build > Check`). Report in the collaborative document the messages you receive about dependencies. Address the messages by fixing the code, and re-run `devtools::check()`. Iterate until there are no more messages about dependencies.
```
Comment (bvreede) [inline _episodes/07-dependencies.md:117]:
```suggestion
> Run `devtools::check()` (or `Build > Check`). What messages do you get about dependencies?
> Optionally, you can address the messages by changing the code, and re-run `devtools::check()` to see if you were successful. 
```

**Review by bvreede — CHANGES_REQUESTED** [review]
Thanks so much!!
It's really nice to have this updated in the Dependencies chapter; it really needs more meat :)

I wonder if you could look again at the new function you introduce. I think it may be worth changing it to be a little simpler to understand (reshaping is not the easiest subject to grasp right away if you've never thought about this), and also to have some more meaning around introducing this new function.

I would also vote for omitting the pipe at this point. We do discuss this in the course, but I don't think it needs to be in the materials at this point, and may make things needlessly complicated. But perhaps @liekelotte or @PabRod have different ideas?

Thanks again! 🙏🏼🙏🏼

File: _episodes/07-dependencies.md [hunk 2]
```diff
@@ -90,4 +90,64 @@ More specifically, we can see that a version equal or higher than `3.0.0` is sug
...
+>~~~r
+> reshape_groups <- function(names) {
+>  df <- data.frame(make_groups(names))
+>  names(df) <- c("group1", "group2")
+>  out <- df %>% tidyr::pivot_longer(cols = tidyselect::starts_with("group"))
+>  return(out)
+>}
+>~~~
+> {: .code}
+>
```
[the remark this answers is not in the document]
Comment (f-hafner) [reply _episodes/07-dependencies.md:116]: reshape was the first example I could come up with, but I agree that simpler is better; I'll use `mutate`. And I omit the pipe for now, too.

[the remark this answers is not in the document]
Comment (PabRod) [reply _episodes/07-dependencies.md:116]: I agree with @bvreede, a simpler function may help. But I don't see this as very critical. If this is slowing you down, we can merge it as it is and take care of the function at a later iteration.

[the remark this answers is not in the document]
Comment (f-hafner) [reply _episodes/07-dependencies.md:116]: no worries @PabRod, I will push a simpler example next week. It's almost done but I haven't had time to finish it

**Comment by f-hafner** [comment]
Hi @bvreede @PabRod , the example is now simplified. Let me know what you think.
Also, when trying it out on my machine, I got this warning: 
```R
❯ checking for unstated dependencies in examples ... OK
   WARNING
  ‘qpdf’ is needed for checks on size reduction of PDFs
```
I think it is unrelated to the changes I made, but let me know if I should check this again.

**Pull request merged**